includes/rest/class-inbox-controller.php

```diff
@@ -284,13 +288,28 @@ public function get_item_schema() {
 	 * @return array An array of user IDs who are the recipients of the activity.
 	 */
 	private function get_local_recipients( $activity ) {
+		// Public activity, deliver to all local ActivityPub users.
+		if ( is_activity_public( $activity ) ) {
+			return Actors::get_all_ids();
```

Comment:
I think you tried to explain this to me before, if that's the case please bear with me.

If the activity is public it might not have followers defined as its audience? Is that why it goes to everyone?
Would we run into a situation where User A subscribed to the actor for this activity and now User B gets those posts in their timeline?